src/Gui/ViewProviderGroupExtension.cpp

```diff
@@ -28,19 +28,66 @@
 
 #include <App/Document.h>
 #include <App/DocumentObject.h>
+#include <App/DocumentObjectGroup.h>
 #include <App/GroupExtension.h>
 #include <Base/Console.h>
 #include <Base/Tools.h>
 
 #include "ViewProviderGroupExtension.h"
 #include "ViewProviderDocumentObject.h"
+#include "Application.h"
 #include "Command.h"
 #include "Document.h"
 #include "MainWindow.h"
 
 
 using namespace Gui;
 
+namespace {
+    // helper function to recursively delete group contents while respecting view provider onDelete methods
+    void deleteGroupContentsRecursively(App::GroupExtension* group, ViewProvider* groupViewProvider) {
```

Comment:
unused parameter 'groupViewProvider' [clang-diagnostic-unused-parameter]
```suggestion
    void deleteGroupContentsRecursively(App::GroupExtension* group, ViewProvider*) {
```